include/seastar/core/metrics.hh

```diff
@@ -358,16 +364,19 @@ struct metric_definition_impl {
     metric_function f;
     description d;
     bool enabled = true;
+    std::vector<std::string> aggregate_labels;
```

Comment:
I think this should be `sm::label` instead of `std::string`.

Reply:
I'm not too sure. `metric_definition_impl` unpacks the `sm::label_instance` objects that it receives into a string-to-string map, so from that point of view using the stronger type doesn't seem consistent with the existing code. You'd also have to unpack the `sm::label` when you do the aggregation, but that's not a big deal.

That being said, I think it would make sense use `sm::label` instead of `std::string` in the public interface and everywhere "above" `metric_definition_impl`.